enhancements/declarative-index-config.md

```diff
@@ -0,0 +1,426 @@
+
+### Goals
+
+A human consumable representation of an index in the form of json/yaml will: 
+
+1. Allow for validation of an index against a standard definition of an index (possibly using a data definition language)
+2. Provide a way for index authors to reason about their indexes via a visual representation of the index, without having to curl a grpc api that in turn queries a sqllite database underneath etc
+3. Provide a way for tools like `opm`, `operator-registry` etc to consume the representation of an index instead of needing a sqlite database as an input whenever they need information about the index to perform a task. 
+4. Allow for reproducability of an index using just the yaml/json representation   
+
+### Non-Goals
+
+
+### User Stories [optional]
+
+#### Story 1
+
+As an index author, I can query an index for a json/yaml representation of the contents of an index. 
+
+For example, if an index `docker.io/my-namespace/my-index` has one bundle `quay.io/etcd` added to it,
+
+`opm represent --output=json docker.io/my-namespace/my-index` will output the following json:
+
+```output.json
+{
+    "graph": {
+        "nodes": [
+            {
+                "bundlePath": "quay.io/etcd",
+                "provides": [
+                    {
+                        "type": "olm.package",
+                        "version": "1.0.0",
+                        "packageName": "etcd"
+                    },
+                    {
+                        "type": "olm.gvk",
+                        "version": "v1beta2",
+                        "group": "etcd.database.coreos.com",
+                        "kind": "EtcdCluster"
+                    },
+                    {
+                        "type": "label",
+                        "value": "alpha"
+                    }
+                ],
+                "constraints": [
+                    [
+                        {
+                            "type": "label",
+                            "value": "LTS"
+                        }
+                    ]
+                ]
+            },
+        ]
+    }
+}
+```
+
+#### Story 2
+
+As an index author, I can validate the contents and structure of my index using the json representation I may retrieve from the above step.
+
+For example, if I have the above `output.json`, I can validate my index by using 
+
+`opm index validate output.json <index-definition>`
+
+where the <index-definition> is a definition of an index that lives in the [api repo](https://github.com/operator-framework/api),using a data definition language.
```

Comment:
nit: I think `<index-definition>` needs to be wrapped in a ` `` `  because the word just disappears in the markdown view. This sentence reads:

where the is a definition of an index

I think markdown might see it as an invalid HTML tag and doesn't render it